WebDriverAgentLib/Categories/XCUIElement+FBFind.m

```diff
@@ -13,21 +13,32 @@
 #import "FBElementTypeTransformer.h"
 #import "XCElementSnapshot.h"
 #import "XCElementSnapshot+FBHelpers.h"
+#import "XCUIElement+FBUtilities.h"
 #import "XCUIElement+FBWebDriverAttributes.h"
 
 @implementation XCUIElement (FBFind)
 
 
 #pragma mark - Search by ClassName
 
-- (NSArray<XCUIElement *> *)fb_descendantsMatchingClassName:(NSString *)className
+- (NSArray<XCUIElement *> *)fb_descendantsMatchingClassName:(NSString *)className shouldReturnAfterFirstMatch:(BOOL)shouldReturnAfterFirstMatch
 {
   NSMutableArray *result = [NSMutableArray array];
   XCUIElementType type = [FBElementTypeTransformer elementTypeWithTypeName:className];
   if (self.elementType == type || type == XCUIElementTypeAny) {
     [result addObject:self];
   }
-  [result addObjectsFromArray:[[self descendantsMatchingType:type] allElementsBoundByIndex]];
+  XCUIElementQuery *query = [self descendantsMatchingType:type];
+  if (shouldReturnAfterFirstMatch) {
+    if (0 == [result count]) {
```

Comment:
You can simplify this logic by making quick return. Meaning checking for `shouldReturnAfterFirstMatch` and `result.count` before you call `descendantsMatchingType` and return if you found something.